### PR TITLE
Remove irrelevant tests and files,  refactor processor request

### PR DESCRIPTION
**Pull request by nozim**
- replace manually created mock objects with moq library alternative 
- refactor metamorph server test
- remove context.Context from ProcessorRequest as per context usage best practices 
https://go.dev/blog/context-and-structs

### CODE REVIEW

File: metamorph/processor_request.go
Comment (boecklim) [inline metamorph/processor_request.go:1]: I think this type and function was in the right place here as the other file is called `interface.go`

File: metamorph/processor_request_test.go
Comment (boecklim) [inline metamorph/processor_request_test.go:1]: This test was not super useful, but also didn't hurt. Without it 1 line less is test covered

File: metamorph/processor_test.go [hunk 1]
```diff
@@ -621,15 +621,14 @@ func BenchmarkProcessTransaction(b *testing.B) {
 
 		txs[txID] = &txHash
 
-		processor.ProcessTransaction(NewProcessorRequest(
-			context.Background(),
-			&store.StoreData{
+		processor.ProcessTransaction(&ProcessorRequest{
```
Comment (boecklim) [inline metamorph/processor_test.go:624]: Why not use the constructor as before `NewProcessorRequest`

Comment (nozim) [reply metamorph/processor_test.go:624]: the constructor does not actually do anything except initialising structure and the arguments seem less readable. 
Also you don't have to specify default value for channels for example.

Comment (nozim) [reply metamorph/processor_test.go:624]: also from 
https://pkg.go.dev/context:
"Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:"

File: metamorph/processor_mock.go
Comment (boecklim) [inline metamorph/processor_mock.go:1]: As this file is generated it should not be edited by hand
To update this file use `make install_gen` and `make gen_go`

**Review by boecklim — APPROVED** [review]

**Pull request merged**